### ISSUE TITLE
dash breaks configuration scripts because it is incompatible

### ISSUE THREAD

**Issue by marcopeereboom** [void-linux/void-packages#10949]
### System

* xuname:  
  * Void 4.19.34_1 x86_64 GenuineIntel uptodate rFF
* package:  
  * dash

### Expected behavior

When configuring riscv-pk configure bombs when using dash. If you don't pay attention you don't notice that there are errors since the command does appear to finish. The failure, however, causes compilation to fail later. The smallest repo to reproduce this with is github.com/riscv-pk.

Correct run using ksh:
```
$ ksh ../configure --prefix=$RISCV --host=riscv64-unknown-linux-gnu
checking build system type... x86_64-pc-linux-gnu
checking host system type... riscv64-unknown-linux-gnu
checking for riscv64-unknown-linux-gnu-gcc... riscv64-unknown-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether riscv64-unknown-linux-gnu-gcc accepts -g... yes
checking for riscv64-unknown-linux-gnu-gcc option to accept ISO C89... none needed
checking for riscv64-unknown-linux-gnu-g++... riscv64-unknown-linux-gnu-g++
checking whether we are using the GNU C++ compiler... yes
checking whether riscv64-unknown-linux-gnu-g++ accepts -g... yes
checking for riscv64-unknown-linux-gnu-ar... riscv64-unknown-linux-gnu-ar
checking for riscv64-unknown-linux-gnu-ranlib... riscv64-unknown-linux-gnu-ranlib
checking for riscv64-unknown-linux-gnu-readelf... riscv64-unknown-linux-gnu-readelf
checking for riscv64-unknown-linux-gnu-objcopy... riscv64-unknown-linux-gnu-objcopy
checking for a BSD-compatible install... /usr/bin/install -c
checking how to run the C preprocessor... riscv64-unknown-linux-gnu-gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
configure: configuring default subproject : pk
configure: configuring default subproject : bbl
configure: configuring default subproject : softfloat
configure: configuring default subproject : dummy_payload
configure: configuring default subproject : machine
configure: configuring default subproject : util
configure: creating ./config.status
config.status: creating pk.mk
config.status: creating bbl.mk
config.status: creating softfloat.mk
config.status: creating dummy_payload.mk
config.status: creating machine.mk
config.status: creating util.mk
config.status: creating Makefile
config.status: creating config.h
config.status: config.h is unchanged
```

### Actual behavior

Configure fails like this:
```
$ sh ../configure --prefix=$RISCV --host=riscv64-unknown-linux-gnu 
checking build system type... x86_64-pc-linux-gnu
checking host system type... riscv64-unknown-linux-gnu
checking for riscv64-unknown-linux-gnu-gcc... riscv64-unknown-linux-gnu-gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether riscv64-unknown-linux-gnu-gcc accepts -g... yes
checking for riscv64-unknown-linux-gnu-gcc option to accept ISO C89... none needed
checking for riscv64-unknown-linux-gnu-g++... riscv64-unknown-linux-gnu-g++
checking whether we are using the GNU C++ compiler... yes
checking whether riscv64-unknown-linux-gnu-g++ accepts -g... yes
checking for riscv64-unknown-linux-gnu-ar... riscv64-unknown-linux-gnu-ar
checking for riscv64-unknown-linux-gnu-ranlib... riscv64-unknown-linux-gnu-ranlib
checking for riscv64-unknown-linux-gnu-readelf... riscv64-unknown-linux-gnu-readelf
checking for riscv64-unknown-linux-gnu-objcopy... riscv64-unknown-linux-gnu-objcopy
checking for a BSD-compatible install... /usr/bin/install -c
checking how to run the C preprocessor... riscv64-unknown-linux-gnu-gcc -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... yes
../configure: 4090: test: x: unexpected operator
configure: configuring default subproject : pk
configure: configuring default subproject : bbl
../configure: 4235: test: x: unexpected operator
../configure: 4282: test: x: unexpected operator
configure: configuring default subproject : softfloat
configure: configuring default subproject : dummy_payload
configure: configuring default subproject : machine
configure: configuring default subproject : util
configure: creating ./config.status
config.status: creating pk.mk
config.status: creating bbl.mk
config.status: creating softfloat.mk
config.status: creating dummy_payload.mk
config.status: creating machine.mk
config.status: creating util.mk
config.status: creating Makefile
config.status: creating config.h
config.status: config.h is unchanged
```

As you can see there are now 3 errors in the output yet configure seems to complete. 

One of the offending lines is: `if test "x$enable_print_device_tree" == "xyes"; then :
`

This is easily repdodicible in other configure scripts as well.

### Steps to reproduce the behavior

```
git clone https://github.com/riscv/riscv-pk.git
cd riscv-pk
mkdir build && cd build
../configure --prefix=$RISCV --host=riscv64-unknown-linux-gnu
```

Easiest is probably to trash dash.

**Comment by maxice8** [comment]
POSIX does not define == which dash follows. Fix your scripts to explicitly call bash or another shell that supports == or use POSIX syntax.

**Comment by marcopeereboom** [comment]
Not my script. These are `configure` scripts that are used left and right. I have seen a dozen of them fail. Since you don't want to fix it the workaround is `sudo ln -sf /bin/ksh /bin/sh` to disable dash. No one cares if it is POSIX compatible if it breaks half the `configure` scripts on the internet.

**Comment by maxice8** [comment]
Let me rephrase:

Fix the scripts you use and the people you apparently get your stuff from so that they don't call /bin/sh when using features not in POSIX, failure when using non-POSIX features when calling /bin/sh, defined to be a POSIX compliant shell, is the expected result.

[Ubuntu already did the work of getting backlash for making dash be /bin/sh](https://bugs.launchpad.net/ubuntu/+source/dash/+bug/61463) and we are not moving backwards.

**Comment by maxice8** [comment]
As a sidenote if you are running ./configure on your system and not on xbps-src you can use `xbps-alternatives -s bash`

**Comment by maxice8** [comment]
Should be fixed upstream once https://github.com/riscv/riscv-pk/pull/157 is merged

**Comment by Anachron** [comment]
It‘s merged upstream now.

**Comment by marcopeereboom** [comment]
I have the need to build from source while developing things. This isn't just riscv-pk, gcc also fails with a few of these (ran into this while building a cross compiler). I appreciate being POSIX correct but that test is done in many configure scripts and I don't have the time to go read every single configure output line to see if someone uses test wrong.

**Comment by maxice8** [comment]
Then set bash your /bin/sh with `# xbps-alternatives -s bash` while dealing with such broken scripts.

Using non-posix features when explicitly calling a posix shell is obviously not supported.

**Comment by maxice8** [comment]
Also there is stuff like `shellcheck` nowadays which always warns you of bashisms on /bin/sh files, there is no excuse for failing to stick to posix sh.

**Comment by marcopeereboom** [comment]
I know of all those tools but these are not my repos. My linux basically broke with a bunch of config scripts overnight. The repo you sent a PR two has two more of these tests:
```
../configure: 4235: test: x: unexpected operator
../configure: 4282: test: x: unexpected operator
```
I'll stop complaining but it would be great if I didn't have to ln it to a working shell. It can't be deleted with the package manager because base depends on it. It is pretty silly that I have to ln ksh to sh.

Sorry but dash is basically broken for anyone who deals with configure or old scripts.

**Comment by maxice8** [comment]
just set bash to /bin/sh with `# xbps-alternatives -s bash`.

It would be the greatest if you stopped wasting everyone's time wanting a defined POSIX shell to support NON-POSIX features (an issue we fixed back in 2006) and worked with the projects you use to fix their **obviously** broken scripts, the simplest fix is to call /bin/bash or /bin/ksh explicitly instead of calling /bin/sh.

> Sorry but dash is basically broken for anyone who deals with configure or old scripts.

We have 2313 packages that use gnu autoconf scripts (/bin/sh) and 89 packages that are self-made configure scripts.
